### PR TITLE
Line labels plugin runs in plugin bar

### PR DESCRIPTION
**Pull request by ibusko**
This is a cleaner version of the older PR #646. The commit history was re-made into a single comit, so we could get rid of the many non-significant commits (resulting from merges from upstream) that refused to go away when rebasing.

I suspect this was caused by something broken in the fork where #646 comes from, because other PRs from that fork also showed the same behavior. 

The code in here is a plain copy of the same code that was already reviewed in #646. Please refer to #646 to access the code review and previous conversation. The original, first PR in this series, is #601.

### CODE REVIEW

**Comment by nmearl** [comment]
To get rid of some of the docs build errors, be sure to remove the module references in the `specviz/docs/api.rst`. For instance, `specviz.core.annotation` can be removed, along with some others.

**Comment by ibusko** [comment]
@nmearl It seems I got rid of the doc build errors. The remaining build errors are more difficult to figure out. 

For instance, the build_docs error apparently is about icon files that can't be found, even though they are in the resources directory. I wonder if that has to do with the new Label-48.png icon in that directory. I found that pytest-qt can't handle png icons. Seems unrelated, but maybe something similar is happening here.

The other two build tests end in core dumps. It's hard to track withou access to the CI environment. Do you have any suggestions?

**Comment by ibusko** [comment]
@nmearl I don't think I ever set up the proper environment to run CI tests. How do I set it up?

**Comment by nmearl** [comment]
Run `pytest specviz` in your `specviz` directory.

**Comment by ibusko** [comment]
I always get an empty run terminating with Abort trap: 6

```
(specviz_dev) bash-3.2$ pytest specviz_stscissb
======================================= test session starts =======================================
platform darwin -- Python 3.7.0, pytest-3.5.1, py-1.7.0, pluggy-0.6.0
PyQt5 5.11.2 -- Qt runtime 5.11.1 -- Qt compiled 5.11.1
rootdir: /Users/busko/Projects/specviz_stscissb, inifile: setup.cfg
plugins: remotedata-0.3.0, qt-3.0.0, openfiles-0.3.0, doctestplus-0.1.3, arraydiff-0.2
collecting 32 items / 3 errors                                                                    Abort trap: 6
(specviz_dev) bash-3.2$ 
```

**Comment by nmearl** [comment]
You may need to fix the icon issue first.

**Comment by ibusko** [comment]
When fixing the icon, I get this:

```
(specviz_dev) bash-3.2$ pyrcc5 specviz/data/resources/resources.qrc -o specviz/widgets/resources.py
Cannot find file: specviz.icns
(specviz_dev) bash-3.2$ 

```

**Comment by nmearl** [comment]
Weird, add this file to your `specviz/data/resources` directory: https://stsci.box.com/s/6nd6kkjk3ufmxbi8w2y566a8vkqqh0fv

**Comment by ibusko** [comment]
Box says the file was removed or is not available.

**Comment by nmearl** [comment]
Try the link now.

**Comment by nmearl** [comment]
When I use a `.svg` file instead of the `.png`, all tests pass. I'm not sure what the issue is about, but I suggest you just use this icon: https://stsci.box.com/s/7ocy6ry87ls7tvjo1hyzvyalbmv0uxpw

**Comment by ibusko** [comment]
That's weird. Thanks for finding this. At least now we have a price tag and can charge cash for specviz :-)

Now there is only one doc test still failing. It fails because warnings are turned into errors by the -W flag in sphinx. When I run it locally with no -W flag, it passes. Although there are 392 warnings, all associated to the plotting.py module, as in "toctree contains reference to nonexisting document 'api/specviz.widgets.plotting.QMdiSubWindow". 

The end product of that run (the html page) suggests that sphinx is attempting to treat Qt classes as specviz classes because it shows some of the Qt classes in the output.

**Comment by nmearl** [comment]
😆 

> It fails because warnings are turned into errors by the -W flag in sphinx.

This is intentional. Warnings indicate that the docs were not built or not built correctly and are intended to block merges.

Since the plotting module api docs built before this PR, it seems we're still introducing something weird in this case. Paging @astrofrog.

File: specviz/plugins/line_labels/annotation.py [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+
+from qtpy.QtCore import QPointF, QRectF
+from qtpy.QtGui import QPolygonF, QPen, QColor
+
+orientations = {
+    'horizontal': {'anchor': (0.5, 1), 'angle': 0},
+    'vertical': {'anchor': (0, 0.5), 'angle': 90}
+}
+
+
```
Comment (astrofrog) [inline specviz/plugins/line_labels/annotation.py]: This file is missing an ``__all__``

File: specviz/plugins/line_labels/line_labels_plotter.py [hunk 2]
```diff
@@ -1,89 +1,135 @@
...
+from astropy import units as u
 
-__all__ = ['LineLabelsPlotter', 'ZoomMarkersThread', 'ZoomEventBuffer']
+from .annotation import LineIDMarker, LineIDMarkerProxy
+from .linelist import LineList, WAVELENGTH_COLUMN, \
+    REDSHIFTED_WAVELENGTH_COLUMN, \
+    MARKER_COLUMN, ID_COLUMN, COLOR_COLUMN, HEIGHT_COLUMN
 
 
 class LineLabelsPlotter(object):
```
Comment (astrofrog) [inline specviz/plugins/line_labels/line_labels_plotter.py]: This file is also missing an ``__all__``

File: specviz/plugins/line_labels/linelists_window.py [hunk 3]
```diff
@@ -47,9 +45,20 @@
 PLOTTED = "Plotted"
```
Comment (astrofrog) [inline specviz/plugins/line_labels/linelists_window.py]: Missing ``__all__``

File: specviz/widgets/plotting.py [hunk 4]
```diff
@@ -17,18 +17,16 @@
...
-from ..core.linelist import LineList, WAVELENGTH_COLUMN, ID_COLUMN
-from .line_labels_plotter import LineLabelsPlotter
 
 __all__ = ['PlotWindow', 'PlotWidget']
 
 
 class PlotWindow(QMdiSubWindow):
     """
-    Displayed plotting subwindow available in the QMdiArea.
+    Displayed plotting subwindow available in the `QMdiArea`.
```
Comment (astrofrog) [inline specviz/widgets/plotting.py]: Don't add backticks here - this is what's causing the error. Single backticks mean a proper API reference. Use double backticks if you want it to look like code.

Comment (ibusko) [reply specviz/widgets/plotting.py]: Done.

File: specviz/plugins/line_labels/linelists_window.py (hunk 3 above)
Comment (ibusko) [reply specviz/plugins/line_labels/linelists_window.py]: Done.

File: specviz/plugins/line_labels/line_labels_plotter.py (hunk 2 above)
Comment (ibusko) [reply specviz/plugins/line_labels/line_labels_plotter.py]: Done.

File: specviz/plugins/line_labels/annotation.py (hunk 1 above)
Comment (ibusko) [reply specviz/plugins/line_labels/annotation.py]: Done.

**Pull request merged**